src/Controllers/DatabaseController.js

```diff
@@ -1365,6 +1365,32 @@ class DatabaseController {
       });
   }
 
+  objectToEntriesStrings(query: any): Array<string> {
+    return Object.entries(query).map(a => a.map(s => JSON.stringify(s)).join(':'));
+  }
+
+  reduceOrOperation(query: { $or: Array<any> }): any {
+    const queries = query.$or.map(q => this.objectToEntriesStrings(q));
+    for (let i = 0; i < queries.length - 1; i++) {
+      for (let j = i + 1; j < queries.length; j++) {
+        const [shorter, longer] = queries[i].length > queries[j].length ? [j, i] : [i, j];
+        const foundEntries = queries[longer].reduce(
+          (acc, entry) => acc + (queries[shorter].includes(entry) ? 1 : 0),
+          0
+        );
+        if (foundEntries === queries[shorter].length) {
+          // If the shorter query is completely contained on the longer one, we can skip it.
+          query.$or.splice(longer, 1);
+          queries.splice(longer, 1);
+        }
+      }
+    }
+
+    if (query.$or.length === 1) return query.$or[0];
```

Comment:
Can you please wrap the if block in brackets?

@dplewis Shouldn't lint currently enforce this? If not, I think we should add that rule; if blocks without brackets are just an unnecessary bug risk IMHO.